tensorflow/core/nccl/nccl_manager.cc

```diff
@@ -533,7 +533,7 @@ void NcclManager::RunCollective(Collective* collective) {
       // Wait to ensure that the kernel that produces the data in the input
       // tensor has finished running before the nccl kernel runs on the
       // communication stream.
-      nccl_stream->stream->ThenWaitFor(p->tensor_stream);
+      nccl_stream->stream->ThenWaitFor(p->input_event.get());
```

Comment:
Please exclude the corresponding commit from this PR. It is too significant for a cherry-pick.

Reply:
Reverted in https://github.com/tensorflow/tensorflow/pull/32474/commits/41041a9c7920a8eeaad0f8feff2838649b49d5ac